application/shared-webapp/ui/components/Toast.tsx

```diff
@@ -94,6 +119,56 @@ function Toast<T extends ToastContents>({ state, ...props }: Readonly<ToastProps
   const { toastProps, titleProps, closeButtonProps, descriptionProps } = useToast(props, state, ref);
   const { content } = props.toast;
 
+  // Track if the toast is being hovered or focused
+  const [isPaused, setIsPaused] = useState(false);
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+  const remainingTimeRef = useRef<number | null>(null);
+  const startTimeRef = useRef<number | null>(null);
+
+  // Handle auto-dismiss functionality with pause on hover/focus
+  useEffect(() => {
+    // Only set up auto-dismiss if content is a ToastOptions object with a duration
+    if (
+      content &&
+      typeof content === "object" &&
+      !isValidElement(content) &&
+      "duration" in content &&
+      typeof content.duration === "number"
+    ) {
+      // If we're not paused, start/resume the timer
+      if (!isPaused) {
+        // If we have remaining time from a pause, use that, otherwise use the full duration
+        const duration = remainingTimeRef.current !== null ? remainingTimeRef.current : content.duration;
+        startTimeRef.current = Date.now();
+
+        timerRef.current = setTimeout(() => {
+          toastQueue.close(props.toast.key);
+        }, duration);
+      } else if (timerRef.current && startTimeRef.current !== null) {
+        // If we're paused and have an active timer, clear it and calculate remaining time
+        clearTimeout(timerRef.current);
+        timerRef.current = null;
+
+        // Calculate how much time is left
+        const elapsedTime = Date.now() - startTimeRef.current;
+        remainingTimeRef.current = Math.max(0, content.duration - elapsedTime);
+      }
+
+      // Clean up timer if toast is dismissed manually
+      return () => {
+        if (timerRef.current) {
+          clearTimeout(timerRef.current);
+        }
+      };
+    }
+  }, [content, props.toast.key, isPaused]);
+
+  // Event handlers for mouse and keyboard interactions
+  const handleMouseEnter = () => setIsPaused(true);
+  const handleMouseLeave = () => setIsPaused(false);
+  const handleFocus = () => setIsPaused(true);
+  const handleBlur = () => setIsPaused(false);
```

Comment:
We might need to [useCallback](https://react.dev/reference/react/useCallback) avoiding unnecessary rerenders.

(I'll look into adding React Compiler for automatic memorization in the code - it's beta and currently only eslint supports the new hook rules)